### PR TITLE
Adding a custom data class alternative for responses

### PR DESCRIPTION
**Pull request by noahkohrs**
We had issue with componentN() functions while targeting Java (#14).

As an alternative, which avoid reimplementing every functions in each classe, we created an abstract custom Data class called Response.,

### CODE REVIEW

File: riot-api/src/main/kotlin/com.noahkohrs.riot.api/Response.kt
Comment (DNSJambon) [inline riot-api/src/main/kotlin/com.noahkohrs.riot.api/Response.kt:1]: is this sorcery

**Review by DNSJambon — APPROVED** [review]
pas capté ça mais wp

**Comment by DNSJambon** [comment]
ah ok masterclass

**Comment by noahkohrs** [comment]
I don't think it's worth to merge for now since I can't think of many use-cases.

**Pull request closed without merging**